File: waimai/web/src/entity/my-order.ts
import {Customer} from './customer';
import {DishesOrder} from './dishesOrder';
import {Restaurant} from './restaurant';
import {Rider} from './rider';

export class MyOrder {

  id?: number;

  customer?: Customer;

  dishesOrders?: DishesOrder[];

  totalPrice?: number;

  status?: number;

  restaurant?: Restaurant;

  rider?: Rider;

  createTime?: number;


  constructor(data = {} as {
    id?: number,
    customer?: Customer,
    dishesOrders?: DishesOrder[],
    totalPrice?: number,
    status?: number,
    restaurant?: Restaurant,
    rider?: Rider,
    createTime?: number
  }) {
    this.id = data.id;
    this.customer = data.customer;
    this.dishesOrders = data.dishesOrders;
    this.totalPrice = data.totalPrice;
    this.status = data.status;
    this.restaurant = data.restaurant;
    this.rider = data.rider;
    this.createTime = data.createTime;
  }
}

File: api/src/main/java/club/yunzhi/log/entity/MyOrder.java
package club.yunzhi.log.entity;


import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@Entity
public class MyOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    private Customer customer;

    @OneToMany
    private List<DishesOrder> dishesOrders;

    private Float totalPrice;

    private Long status;

    @OneToOne
    private Restaurant restaurant;

    @OneToOne
    private Rider rider;

    @CreationTimestamp
    private Timestamp createTime;
}
